### PR TITLE
refactor: Add types to some tests

### PR DESCRIPTION
**Pull request by marcuslimdw**
### Pull Request Checklist

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [ ] Pre-Commit Checks were ran and passed
- [ ] Tests were ran and passed

### Description

Add type parameters to some test cases where they were not used; partially fixes https://github.com/litestar-org/litestar/issues/2694.

### CODE REVIEW

File: litestar/types/builtin_types.py [hunk 1]
```diff
@@ -23,3 +24,4 @@
 
 UnionTypes = {UnionType, Union}
 TypedDictClass: TypeAlias = Type[_TypedDictMeta]
+EmptyDict: TypeAlias = Dict[Never, Never]
```
Comment (peterschutt) [inline litestar/types/builtin_types.py:27]: Its probably not worth adding a new library type just to type those tests - I'd be happy enough with `dict[Any, Any]` for them.

File: tests/unit/test_openapi/test_responses.py [hunk 2]
```diff
@@ -42,6 +41,8 @@
 from tests.models import DataclassPerson, DataclassPersonFactory
 from tests.unit.test_openapi.utils import PetException
 
+T = TypeVar("T")
+
```
Comment (peterschutt) [inline tests/unit/test_openapi/test_responses.py:45]: Any particular reason to prefer local definition over the import?

**Review by peterschutt — DISMISSED** [review]
Thanks!

Do you have specific reasoning for targeting these particular tests? There are still hundreds of cases of these that remain both in the tests and in the library, so I feel like some sort of organized approach to completing this task might be in order.

File: tests/unit/test_openapi/test_responses.py (hunk 2 above)
Comment (marcuslimdw) [reply tests/unit/test_openapi/test_responses.py:45]: Two things: `T` is not declared in `litestar.response.base.__all__`, and it seems (a little) weird to me to import an unbounded type variable from a sibling module. I'm not sure what the project convention is though, so happy to revert this if you think it's inappropriate

**Comment by marcuslimdw** [comment]
What I did was set `disallow_any_generics` to `true` locally, and then look for stuff that I was reasonably confident could be changed without requiring in-depth knowledge.

I suggest:

* Generating a list of packages which have `disallow-any-generics` violations
* On the root issue, creating a checklist (not sure what you call those lists of checkboxes) with these packages
* Letting people claim them in comments on the root issue and create PRs to fix those packages, reporting back on the issue when they're done

File: litestar/types/builtin_types.py (hunk 1 above)
Comment (peterschutt) [reply litestar/types/builtin_types.py:27]: Well, we don't have a documented standard on adding new types to the library, however if we did this use case would fall short of it IMO.

Comment (marcuslimdw) [reply litestar/types/builtin_types.py:27]: Got it, will inline it

File: tests/unit/test_openapi/test_responses.py (hunk 2 above)
Comment (peterschutt) [reply tests/unit/test_openapi/test_responses.py:45]: Both of those are valid reasons.

We do often import names that aren't declared in `__all__` into tests b/c we often don't include names in there that we don't want in the public interface yet still want to test.

On point two, if I was writing this module today I wouldn't have imported `T` either. Again, no strict convention, but I would say that the common practice is to not import these.

As a counter argument, minor changes do have a cost in that someone has to review them. So as long as you balance that fact against the changes you propose then I'm happy enough.

**Comment by JacobCoffee** [comment]
Can this be merged (after updates) or should it be closed?

File: tests/unit/test_openapi/test_responses.py [hunk 3]
```diff
@@ -44,6 +43,9 @@
...
+
```
Comment (provinzkraut) [inline tests/unit/test_openapi/test_responses.py:48]:
```suggestion
```

**Review by provinzkraut — APPROVED** [review]

**Review by provinzkraut — APPROVED** [review]
I think this PR has set a new record for longest time between opened and merged :)

**Pull request merged**